packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -104,9 +130,17 @@ export class WorkspaceCommandContribution implements CommandContribution {
                     initialValue: uri.path.base,
                     validate: name => this.validateFileName(name, parent)
                 });
-                dialog.open().then(name =>
-                    this.fileSystem.move(uri.toString(), uri.parent.resolve(name).toString())
-                );
+                dialog.open().then(name => {
+                    this.fileSystem.move(uri.toString(), uri.parent.resolve(name).toString()).then(() => {
+                        let event: IUriChangedEvent = {
+                            factoryId: "code-editor-opener",
+                            options: uri.parent.resolve(uri.path.base).toString(),
+                            newOptions: uri.parent.resolve(name).toString(),
+                            newLabel: name
+                        };
+                        this.onIUriChangedEmitter.fire(event);
```

Comment:
Actually thinking more about this it would make more sense to have FileSytem.move() fire this event and have the event emitter there.
From the workspace command it looks quite odd..